examples/miscellaneous/plot_pipeline_display.py

```diff
@@ -0,0 +1,203 @@
+"""
+=================================================================
+Displaying Pipelines
+=================================================================
+
+The default configuration for displaying a pipeline is `'text'` where
+`set_config(display='text')`.  To visualize the diagram in Jupyter Notebook,
+use `set_config(display='diagram')` and then output the pipeline object.
+
+To see more detailed steps in the visualization of the pipeline, click on the
+steps in the pipeline.
+"""
+
+# %%
+# Illustration of a Basic `Pipeline` with `LogisticRegression`
+###############################################################################
+# This section constructs a basic pipeline with one classifer step and displays
+# its text and visual representation.
+
+from sklearn.pipeline import Pipeline
+from sklearn.linear_model import LogisticRegression
+from sklearn import set_config
+
+steps = [("classifier", LogisticRegression())]
+pipe = Pipeline(steps)
+
+# %%
+# To view the text pipeline, the default is `display='text'`
+set_config(display="text")
+pipe
+
+# %%
+# To visualize the diagram, change to `display='diagram'`
+set_config(display="diagram")
+pipe
+
+# %%
+# Illustration of `Pipeline` and `StandardScaler` and `LogisticRegression`
+###############################################################################
+# This section constructs a pipeline with a preprocessing step and classifier.
+# and displays its visual representation.
+
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn import set_config
+
+steps = [
+    ("preprocessing", StandardScaler()),
+    ("classifier", LogisticRegression()),
+]
+pipe = Pipeline(steps)
+
+# %%
+# To visualize the diagram, change `display='diagram'`
+set_config(display="diagram")
+pipe
+
+# %%
+# Displaying a Pipeline Chaining Multiple Preprocessing Steps & Classifier
+###############################################################################
+# HTML Illustration of `Pipeline` and `PolynomialFeatures`, `StandardScaler`
```

Comment:
here and everywhere else it would be nice to link to the specific classes with  `` :class:`~sklearn.pipeline.Pipeline` `` etc.